lib/elixir/lib/module.ex

```diff
@@ -1854,6 +1911,29 @@ defmodule Module do
     value
   end
 
+  defp preprocess_doc_meta([], _module, _line, map), do: map
+
+  defp preprocess_doc_meta([{key, _} | tail], module, line, map)
+       when key in [:opaque, :defaults, :deprecated] do
+    message = "ignoring reserved documentation metadata key: :#{key}"
+    IO.warn(message, attribute_stack(module, line))
+    preprocess_doc_meta(tail, module, line, map)
+  end
+
+  defp preprocess_doc_meta([{key, value} | tail], module, line, map) when is_atom(key) do
+    validate_doc_meta(key, value)
+    preprocess_doc_meta(tail, module, line, Map.put(map, key, value))
+  end
+
+  defp validate_doc_meta(:since, value) when not is_binary(value) do
+    raise ArgumentError,
+          ":since is a built-in documentation metadata key. " <>
+            "It should be a string representing the version a function, macro, type or " <>
```

Comment:
I think there is something missing in this sentence:

> It should be a string representing the version **in which** a function, a macro, a type or a callback was added

(Btw, I'm not sure if **to which** is the proper preposition, maybe better to ask a native English speaker)